modules/shared/src/main/scala/io/constellationnetwork/env/JarSignature.scala

```diff
@@ -14,8 +14,15 @@ import fs2.{Chunk, Stream}
 object JarSignature {
 
   def jarHash[F[_]: Async: Files]: F[Hash] = {
-    val jarPath = JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI
-    digestOf(Files[F].readAll(Path(jarPath.getPath)))
+    val jarPath = Path(JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI.getPath)
+
+    val bytes =
+      if (jarPath.extName === ".jar")
+        Files[F].readAll(jarPath)
+      else
+        readAllBytesFromDir(jarPath)
```

Comment:
Hmm, as I think about it, this will be non deterministic, as you may have a different content of a directory containing jar file. Hence, it's not a good source for hash calculation, that should be deterministic across builds. Also checking the file type by just its extension does not look good too and may be a source of issues. I'd rather stay with the previous version where you assumed the hash can be calculated. If that fails, return an error.

Reply:
Yes, fail if there's an IO problem is the right approach. Now, what to do in the rare situation you're running directly from class files? I don't think we need to support it for production nodes, but I want to make it a little easier if you need to debug from IDE.
There are a few workarounds: locally change the code to return something and revert it, which is cumbersome, or run the code externally from the jar and attach the IDE as remote debug. Those are not particularly appealing.
The other option in case of failure, is to return some arbitrary value: 0, or a hash of the classes, it won't match the eventual hash of the jar but since is a local run and not a production use, it won't matter much.

Now, if we want to run without the jar AND control the hash value (maybe to join another cluster) we can read the value from a system property in the failure recovery.

Reply:
> I don't think we need to support it for production nodes, but I want to make it a little easier if you need to debug from IDE.

Well, so either you make sure that the code works on dev environment only and production stays the same, or you implement it in a way that works for production nodes too.

> The other option in case of failure, is to return some arbitrary value: 0, or a hash of the classes, it won't match the eventual hash of the jar but since is a local run and not a production use, it won't matter much.

This fix actually will be on production nodes so it may happen on production nodes. So again, either you need to make sure it works on Dev (and maybe Testnet) environments only (check for example localhost usage for Dev environment when joining nodes together. it's forbidden in the production code but it works locally), or you need to make sure it won't break production (so returning 0s isn't a great option there).

Reply:
Checking for env is what I actually need, thanks!